### PR TITLE
feat(split chunks): split chunks should consider chunk combinations

### PR DESCRIPTION
**Pull request by JSerFeng**
## Summary

Fix: #4261 

Add combinations for splitChunks.

Before, every cacheGroup will be a splitted chunk
imagine if module `a` exists in entryA and entryB, but `b` exists in entryA, not in entryB, both `a` and `b` will be splitted to a new chunk, so in the end. there are 3 chunks, entryA, entryB, and shared splitted chunk. EntryB ends up including extra module `b`.

We should use module's combinations as key, for splitted chunks, one cacheGroup may end up with multiple chunks.

Before:
```
key                 value
cacheGroup index 1: { chunks: [entryA, entryB], modules: [a, b] }
```

Now:
```
key  value
ab: { chunks: [entryA], modules: [a, b] }
b:  { chunks: [entryB], modules: [b] }
```

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: crates/rspack_plugin_split_chunks_new/src/plugin/module_group.rs [hunk 1]
```diff
@@ -240,4 +308,70 @@ impl SplitChunksPlugin {
...
+        .entry(count)
+        .and_modify(|set| set.push(chunks.clone()))
+        .or_insert(Vec::new());
+    }
+
+    (chunk_sets_in_graph, single_chunk_sets, chunk_sets_by_count)
+  }
+}
+
+fn is_sub_set<T: std::hash::Hash + std::cmp::Eq>(
```
Comment (hyf0) [inline crates/rspack_plugin_split_chunks_new/src/plugin/module_group.rs:362]: You could use this API https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.is_subset

**Comment by hyf0** [comment]
Well done. This is a wonderful feature.

**Comment by JSerFeng** [comment]
!canary

**Comment by JSerFeng** [comment]
!canary

**Review by ahabhgk — DISMISSED** [review]

**Comment by JSerFeng** [comment]
!bench

**Comment by hardfist** [comment]
it seem benchmark machine is down...

**Review by ahabhgk — APPROVED** [review]

**Comment by f0reclone** [comment]
@JSerFeng when using the latest canary build which includes this fix, the bundle size did go up from 36MB to 48MB.

The splitting seems to be correct (every node module in a new chunk).

**Comment by hardfist** [comment]
!canary

**Pull request merged**